File: board/src/core/api/eventSection.ts
import axios from 'axios';
import { TargetType, StatusLevel } from '~/core/enums'
import { IEventSection } from "~/core/models"

export const RangeBy = async (chamberIds:string[], fromDate: string, toDate: string) => {
  const res = await axios.get('api/event-section/range-by', {
    params: {
      from_date: fromDate,
      to_date: toDate,
      chamber_ids: chamberIds,
    }
  })

  const resData = res.data as {
    id: string
    from_date: string
    to_date: string
    status: StatusLevel
    code:number
    message:string
    chamber_id:string
    substract_id:string
  }[]

  const eventSections: IEventSection[] = resData.map(x => { 
    return {
      id: x.id,
      fromDate: x.from_date,
      toDate: x.to_date,
      status: x.status,
      code:x.code,
      message:x.message,
      chamberId:x.chamber_id,
      substractId:x.substract_id
    }
  })
  return eventSections
}

File: app/mlboard/api/services/event_section.py
from mlboard.core import models as ms
from mlboard.core import queries as qs
from mlboard.core import db
from fastapi import APIRouter, Query
from typing import List, Dict, Any, Optional, Union
from logging import getLogger
from cytoolz.curried import pipe, map
from pydantic import BaseModel
import uuid
from datetime import datetime

router = APIRouter()


class EventSection(BaseModel):
    id: uuid.UUID
    from_date: datetime
    to_date: datetime
    code: str
    status: int
    message: str
    chamber_id: uuid.UUID
    substract_id: str



@router.get('/event-section/range-by')
async def range_by(
    from_date: datetime,
    to_date: datetime,
    chamber_ids: List[uuid.UUID] = Query(None),
):
    async with db.get_conn() as conn:
        rows = await qs.EventSection(conn).range_by(
            chamber_ids=chamber_ids,
            from_date=from_date,
            to_date=to_date,
        )
    return rows
